engine/src/main/kotlin/io/provenance/scope/ContractWrapper.kt

```diff
@@ -38,7 +36,7 @@ class ContractWrapper(
 
     private val constructor = getConstructor(contractClass)
 
-    private val constructorParameters = getConstructorParameters(constructor, records)
+    private val constructorParameters = getConstructorParameters(constructor, records).map { (it as Either.Left<Any>).value }
```

Comment:
what happens if a constructor parameter is indeed a list of messages (and ends up as an `Either.Right` instance)? Wouldn't this blow up? Can we do a when matching on left/right and return the value either way (since constructorParameters is a `List<Any>`)?

Reply:
Yup, will make those changes now. I am admittedly confused on when the left or the right is defined. I thought the benefit of `Either` is to define the value side and the `Nothing` side but in our instance when would it be `Either.Right`? Just trying to better understand the library.

Reply:
Just to clarify my confusion, I had read on these docs https://arrow-kt.io/docs/0.11/apidocs/arrow-core-data/arrow.core/-either/ and saw that `Either` is "right-biased" which makes me question why our `constructorParameters` are on the left side.

Reply:
Pushed up the change

Reply:
I think if you are using Either to return success/error, then the right hand side is expected to be the 'success' side... in this case, I think it is just the fact that a single constructor arg can either be a proto message or a list of proto messages... but since that gets boiled down to a List<Any>, I wonder if the Either could be eliminated actually

Reply:
Yeah I was thinking that too. It seems that when returning the `List<Any>` response it gets wrapped into an `Either` but since we aren't handling any error case it does seem unnecessary?